compiler/rustc_interface/src/interface.rs

```diff
@@ -452,6 +452,15 @@ pub fn run_compiler<R: Send>(config: Config, f: impl FnOnce(&Compiler) -> R + Se
                 config.expanded_args,
             );
 
+            // Even though the session holds the lint store, we can't build the
+            // lint store until the session exists. So do that now.
+            let mut lint_store = rustc_lint::new_lint_store(sess.enable_internal_lints());
```

Comment:
I think this is now the only caller of `Session::enable_internal_lints`, which means its logic can be inlined above, and then the lint store created at the same time as the session. That would enable the lint store to be non-optional.

Reply:
The lint store could be created, but the initialization via `register_lints` would have to wait until after the `Session` is built. And by that stage the lint store is held in `Session` as `Lrc<dyn Any + DynSync + DynSend>`, which is hard to modify and is intended to be a read-only form.

Reply:
Is there a reason it has to be held as `Lrc<dyn Any + DynSync + DynSend>` rather than by its concrete type (with or without `Lrc`)?

Reply:
Yes. `rustc_session` does not (and cannot) depend on `rustc_lint`.

Reply:
As the comment on `lint_store` says:
```
/// This only ever stores a `LintStore` but we don't want a dependency on that type here.
```